common/src/types/token.rs

```diff
@@ -502,10 +133,12 @@ impl Token {
         }
 
         let ticker = self.get_ticker().unwrap();
-        ALL_EXCHANGES
-            .iter()
-            .filter(|&&exchange| exchange != Exchange::UniswapV3)
-            .filter(|&exchange| EXCHANGE_TICKERS.get(exchange).unwrap().contains_key(ticker))
+        EXCHANGE_SUPPORT_MAP
+            .get()
+            .unwrap()
+            .get(ticker)
+            .map_or(vec![], |supported_exchanges| supported_exchanges.keys().collect())
+            .into_iter()
             .for_each(|&exchange| {
                 exchanges.insert(exchange);
             });
```

Comment:
this is a sort of odd way to mutate the `exchanges` value above. I'd suggest we clean up the method as:
```rust
pub fn supported_exchanges(&self) -> HashSet<Exchange> {
    if !self.is_named() {
        return HashSet::from([Exchange::UniswapV3])
    }

    let ticker = self.get_ticker().unwrap();
    get_exchange_support().get(ticker).map(|exchanges| exchanges.keys().collect()).unwrap_or_default()
}
```

Note also that I added a helper `get_exchange_support()`; probably a useful thing to export that unwraps the `OnceLock`